javalib/src/main/scala/java/util/concurrent/CopyOnWriteArrayList.scala

```diff
@@ -397,28 +414,165 @@ class CopyOnWriteArrayList[E <: AnyRef] private (private var inner: js.Array[E])
     if (index < 0 || index > size())
       throw new IndexOutOfBoundsException(index.toString)
   }
+
+  @inline private def ensureCapacity(minCapacity: Int): Unit = {
+    if (isWebAssembly) {
+      if (innerImpl.capacity(inner) < minCapacity) {
+        val newCapacity = roundUpToPowerOfTwo(minCapacity + 1) // Index 0 stores the length
+        inner = innerImpl.resize(inner, minCapacity)
+      }
+    }
+    // We ignore this in JS as js.Array doesn't support explicit pre-allocation
```

Comment:
Like in `PriorityQueue`, we can replace the Wasm-only `capacity()` and `resize()` by a generic, virtual `ensureCapacity()` (which happens to do nothing in `JSArrayImpl`).